api/loaders/mysql_loader.py

```
@@ -45,6 +53,35 @@ class MySQLLoader(BaseLoader):
 
+    @staticmethod
+    def _execute_count_query(cursor, table_name: str, col_name: str) -> Tuple[int, int]:
+        """
+        Execute query to get total count and distinct count for a column.
+        MySQL implementation returning counts from dictionary-style results.
+        """
+        query = f"""
+            SELECT COUNT(*) AS total_count,
+                COUNT(DISTINCT `{col_name}`) AS distinct_count
+            FROM `{table_name}`;
+        """
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)